doc/specs/#658 - WinGet Download.md

```diff
@@ -0,0 +1,156 @@
+---
+author: Roy MacLachlan @RDMaclachlan, Ryan Fu @ryfu-msft
+created on: 2023-02-09
+last updated: 2024-02-15
+issue id: 658
+---
+
+# `Download` command
+
+"For [#658](https://github.com/microsoft/winget-cli/issues/658)"
+
+## Abstract
+
+This spec describes the functionality and high-level implementation design for downloading package installers using the Windows Package Manager.
+
+## Inspiration
+
+This is inspired by customer feedback, and a need for broader application deployments:
+* Customers want to share the installer with an offline device. 
+
+## Solution Design
+
+The `download` command will provide users with the ability to download any installer from a single package. The following command options are available:
+
+```
+  -d,--download-directory      Directory where the installers are downloaded to
+  -m,--manifest                The path to the manifest of the package
+  --id                         Filter results by id
+  --name                       Filter results by name
+  --moniker                    Filter results by moniker
+  -v,--version                 Use the specified version; default is the latest version
+  -s,--source                  Find package using the specified source
+  --scope                      Select install scope (user or machine)
+  -a,--architecture            Select the architecture
+  --installer-type             Select the installer type
+  -e,--exact                   Find package using exact match
+  --locale                     Locale to use (BCP47 format)
+  --ignore-security-hash       Ignore the installer hash check failure
+  --skip-dependencies          Skips processing package dependencies and Windows features
+  --accept-package-agreements  Accept all license agreements for packages
+  --accept-source-agreements   Accept all source agreements during source operations
+  -?,--help                    Shows help about the selected command
+  --wait                       Prompts the user to press any key before exiting
+  --logs,--open-logs           Open the default logs location
+  --verbose,--verbose-logs     Enables verbose logging for winget
+  --disable-interactivity      Disable interactive prompts
```

Comment:
Does this list match the latest? I added custom-header for rest source recently.

Reply:
Added missing arguments.